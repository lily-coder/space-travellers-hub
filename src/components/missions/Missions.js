/* eslint no-unused-vars : "off" */
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import HeaderMission from './HeaderMission.js';
import BodyMission from './BodyMission.js';

const Missions = () => {
  const { missions } = useSelector((state) => state).missions;

  const dispatch = useDispatch();

  return (
    <section>
      <table>
        <HeaderMission />
        < BodyMission />
      </table>
    </section>
  );
};

export default Missions;