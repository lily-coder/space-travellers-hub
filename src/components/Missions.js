/* eslint-disable no-unused-vars */
import { React, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import MissionsList from '../pages/MissionList.js';
import { fetchDataRequest } from '../redux/missions/missions.js';

const Missions = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchDataRequest());
  }, []);
  return (
    <div>
      <MissionsList />
    </div>
  );
};
export default Missions;