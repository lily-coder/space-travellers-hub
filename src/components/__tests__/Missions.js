import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect.js';
import { Provider } from 'react-redux';
import store from '../../redux/configStore.js';
import Missions from '../Missions.js';

describe('Render the missions lists', () => {
  it('Renders the missions correctly', () => {
    const missions = TestRenderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(missions).toMatchSnapshot();
  });
});