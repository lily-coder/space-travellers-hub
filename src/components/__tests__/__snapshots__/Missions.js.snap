// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the missions lists Renders the missions correctly 1`] = `null`;
